.github/workflows/test-linux.yml

```diff
@@ -0,0 +1,122 @@
+        type: boolean
+        default: false
+
+jobs:
+  unit-tests:
+    if: github.repository_owner == 'Qiskit'
+    name: ${{ inputs.runner }}-tests-Python-${{ inputs.python-version }}
+    runs-on: ${{ inputs.runner }}
+    timeout-minutes: 60
+    env:
+      QISKIT_SUPPRESS_PACKAGING_WARNINGS: Y
+      PIP_CACHE_DIR: ${{ github.workspace }}/.pip
+      QISKIT_TEST_CAPTURE_STREAMS: 1
+      HAVE_VISUAL_TESTS_RUN: false
+    strategy:
+      fail-fast: false
+    steps:
+      - uses: actions/checkout@v4
+      - name: Install Rust toolchain
+        uses: dtolnay/rust-toolchain@1.79
+        if: inputs.python-version == '3.9'
+      - name: Set up Python ${{ inputs.python-version }}
+        id: python-outputs
+        uses: actions/setup-python@v5
+        with:
+          python-version: ${{ inputs.python-version }}
+      - name: Cache stestr
+        uses: actions/cache@v4
+        with:
+          key: 'stestr | "${{ runner.os }}" | "${{ inputs.python-version }}" | "${{ github.run_id }}"'
+          restore-keys: |
+            stestr | "${{ runner.os }}" | "${{ inputs.python-version }}"
+            stestr | "${{ runner.os }}"
+            stestr
+          path: .stestr
+      - name: Prepare venv
+        run: |
+          set -e
+          python -m pip install --upgrade pip setuptools wheel
+          python -m venv test-job
+      - name: Install Qiskit from sdist
+        if: ${{ inputs.install-from-sdist }}
+        run: |
+          set -e
+          # Use stable Rust, rather than MSRV, to spot-check that stable builds properly.
+          rustup override set stable
+          source test-job/bin/activate
+          python -m pip install -U pip build
+          python -m build --sdist .
+          python -m pip install -U \
+            -c constraints.txt \
+            -r requirements.txt \
+            -r requirements-dev.txt \
+            dist/qiskit-*.tar.gz
+          # Build and install both qiskit and qiskit-terra so that any optionals
+          # depending on `qiskit` will resolve correctly.
+      - name: Install Qiskit directly
+        if: ${{ !inputs.install-from-sdist }}
+        run: |
+          set -e
+          source test-job/bin/activate
+          python -m pip install -U \
+            -c constraints.txt \
+            -r requirements.txt \
+            -r requirements-dev.txt \
+            -e .
+          # Build and install both qiskit and qiskit-terra so that any optionals
+          # depending on `qiskit` will resolve correctly.
+      - name: Install Optional packages
+        run: |
+          set -e
+          source test-job/bin/activate
+          python -m pip install -r requirements-optional.txt -c constraints.txt
+          python -m pip check
+        if: ${{ inputs.install-optionals }}
+      - name: Install optional non-Python dependencies
+        run: |
+          set -e
+          sudo apt-get update
+          sudo apt-get install -y graphviz
```

Comment:
Do we still need these here? I feel like they can be removed.

Reply:
I think we have regular unit tests that run graphviz for visualization testing so we will need this on jobs that run with optionals enabled.